Source/Scene/processPbrMaterials.js

```diff
@@ -488,6 +544,22 @@ define([
             '#endif \n' +
             '}\n\n';
 
+        fragmentShader +=
+            'vec2 computeTexCoord(vec2 texCoords, vec2 offset, float rotation, vec2 scale, bool repeatS, bool repeatT) \n' +
+            '{\n' +
+            '    mat3 translationMatrix = mat3(1.0, 0.0, 0.0, 0.0, 1.0, 0.0, offset.x, offset.y, 1.0); \n' +
+            '    rotation = -rotation; \n' +
+            '    mat3 rotationMatrix = mat3(\n' +
+            '        cos(rotation), sin(rotation), 0.0, \n' +
+            '       -sin(rotation), cos(rotation), 0.0, \n' +
+            '        0.0, 0.0, 1.0); \n' +
+            '    mat3 scaleMatrix = mat3(scale.x, 0.0, 0.0, 0.0, scale.y, 0.0, 0.0, 0.0, 1.0); \n' +
+            '    vec2 transformedTexCoords = ((translationMatrix * rotationMatrix * scaleMatrix) * vec3(fract(texCoords), 1.0)).xy; \n' +
```

Comment:
Can these three be combined into a single matrix to avoid the matrix multiplication?